rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -912,11 +944,72 @@ void main_window::DecryptSPRXLibraries()
 				{
 					gui_log.error("Failed to create %s", new_path);
 				}
+
+				break;
 			}
-			else
+			else if (!invalid)
 			{
-				gui_log.error("Failed to decrypt %s", old_path);
+				// Allow the user to manually type KLIC if decryption failed
+
+				if (tried)
+				{
+					gui_log.error("Failed to decrypt %s with specfied KLIC, retrying", old_path);
+				}
+
+				const std::string filename = old_path.substr(old_path.find_last_of(fs::delim) + 1);
+
+				input_dialog dlg(32, QString{}, tr("Enter KLIC of %0").arg(qstr(filename)),
+					tr("Hexadecimal only."), "00000000000000000000000000000000", this);
+
+				QFont mono = QFontDatabase::systemFont(QFontDatabase::FixedFont);
+				mono.setPointSize(8);
+				dlg.set_input_font(mono, true, '0');
+				dlg.set_clear_button_state(false);
+				dlg.set_validator(new QRegExpValidator(QRegExp("^[a-fA-F0-9]*$"))); // HEX only
+
+				if (dlg.exec() == QDialog::Accepted)
+				{
+					const std::string text = sstr(dlg.get_input_text());
+
+					if (text.size() == 32)
+					{
+						auto& klic = (tried ? klics.back() : klics.emplace_back());
+
+						// It must succeed (only hex characters are present)
+						std::from_chars(&text[0], &text[16], klic._u64[1], 16); // Not a typo: on LE systems the u64[1] part will be swapped with u64[0] later
+						std::from_chars(&text[16], &text[32], klic._u64[0], 16); // And on BE systems it will be already swapped by index internally
+
+						// Needs to be in big endian because the left to right byte-order means big endian
+						klic = std::bit_cast<be_t<v128>>(klic);
+
+						// Retry with specified KLIC
+						key_it -= +std::exchange(tried, true); // Rewind on second and above attempt
+						gui_log.notice("KLIC entered for %s: %s", filename, klic);
+						continue;
+					}
+					else
```

Comment:
ups. I guess you'll need that connect to textChanged after all in order to only allow the ok button enabled if the text has max length.
Doesn't make much sense to accept a dialog that's about to fail anyway...

Reply:
How would I disable only the "Okay" button and not the cancel? or do you want to hide the OK button completely in this case?

Reply:
Try something like this and only enabled/disable QDialogButtonBox::StandardButton::Ok in the textChanged connect.
Of course it should also be disabled in the beginning.
```
void input_dialog::set_button_enabled(QDialogButtonBox::StandardButton id, bool enabled)
{
	if (QPushButton* button = m_button_box->button(id))
	{
		button->setEnabled(enabled);
	}
}
```

Reply:
Implemented.